### ISSUE TITLE
Doesn't work with dynamically loaded content

### ISSUE THREAD

**Issue by jacobwhite** [pelmers/moz-text-rewriter#7]
Doesn't work on content that loads after the initial page load.

**Comment by MathiasRenner** [comment]
Thanks for the info, saved me some minutes of trying it out myself :-)

**Comment by MathiasRenner** [comment]
I confirm: Does not change dynamically created content.

**Comment by pelmers** [comment]
thanks for report! this bugs me sometimes too. Some day once it bugs me enough I'll add this feature!

Most likely through MutationObservers.

**Comment by jacobwhite** [comment]
I ended up making my own for my specific use case using a mutationobserver

**Comment by pelmers** [comment]
nice! I put something quick together last night too, expect an update soon.

**Comment by pelmers** [comment]
updated version is out on the Firefox and Chrome addon websites pending review!

fixed in 58ef04358c989e

**Comment by MathiasRenner** [comment]
@pelmers Thanks a lot for your ast respone, awesome! However it does not work for me yet.
Use case: In this URL: https://www.qwant.com/?q=test&t=web, I'd like to delete all  onmousedown listeners like these ones:

![grafik](https://user-images.githubusercontent.com/2377464/43865113-dfade744-9b61-11e8-8731-2107827a9580.png)

This is my rule:
![grafik](https://user-images.githubusercontent.com/2377464/43865172-13fd152e-9b62-11e8-837f-9cf19fceae4e.png)

Did I configure something wrong?

**Comment by pelmers** [comment]
@MathiasRenner  this addon only replaces text content, it doesn't rewrite the page's HTML.

I don't plan to add that, but if you were interested, a starting point would be to change the tree walker to NodeFilter.Element (https://github.com/pelmers/text-rewriter/blob/master/data/text-rewriter.js#L104) and the nodeValue to innerHTML (https://github.com/pelmers/text-rewriter/blob/master/data/text-rewriter.js#L117-L118).

**Comment by jacobwhite** [comment]
That’s what greasemonkey is for.

**Comment by MathiasRenner** [comment]
@jacobwhite  Thanks for the hint. I tried Greasemoney - with the help of a friend we figured out that this is a more complex problem than I thought. I cannot just change the onmousedown-listeners.  I'd have to copy the listeners and insert them again, which will disable them (https://stackoverflow.com/questions/19469881/remove-all-event-listeners-of-specific-type).
I am not fast enough in Javascript, so I've put this on my TODO list for some time later.
